### PR TITLE
Ensure consistent light state icon brightness

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

The coloring was off. Some did not use the RGB colors at all, others ignored the brightness.

Now only the tile card is off, where honestly I have no clear idea what it should do in this case design-wise, so I left that one alone.

Before
image

After
image

Also note that in the "light off" state the tile card is also inconsistent, which also looks like a bug for the end user.
image

@piitaya @matthiasdebaat Can you please comment on the tile card points? I think consistency is really key here (more important than any actual used colors IMHO).

There is also a similar, although more narrow PR I just saw: https://github.com/home-assistant/frontend/pull/14724

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/hui-entity-card.ts [hunk 1]
```diff
@@ -196,13 +197,24 @@ export class HuiEntityCard extends LitElement implements LovelaceCard {
...
   }
 
+  private _computeBrightness(stateObj: HassEntity | LightEntity): string {
+    if (stateObj.attributes.brightness && stateActive(stateObj)) {
+      const brightness = stateObj.attributes.brightness;
+      return `brightness(${(brightness + 245) / 5}%)`;
+    }
+    return "";
+  }
+
```
Comment (spacegaier) [inline src/panels/lovelace/cards/hui-entity-card.ts:217]: I don't really like this code duplication, but was already present for the `_computeColor` as well, so did not change that in this PR.

**Comment by piitaya** [comment]
There is some changes for the tile card : 
- no brightness because not only the icon is colored but also the shape and the slider (if set)
- some transformations on the RGB value to avoid this case : https://github.com/home-assistant/frontend/issues/14696

We haven't discuss this point. What do you think?

**Comment by spacegaier** [comment]
> * no brightness because not only the icon is colored but also the shape and the slider (if set)

Which means one cannot really use the tile card next to other existing UI elements since that looks weird/inconsistent. So a user either will have to build a dashboard with only tile cards or only non-tile cards.


>     * some transformations on the RGB value to avoid this case : [ICON State Color for colored light tracks color when light is white #14696](https://github.com/home-assistant/frontend/issues/14696)

Yes that was always an issue, which is one reason why I use a dark theme 😆 so that never comes into play (plus a lot less power consumption from the wall tablets).

> There is also some weird think about rgb color on entity card : RGB value are displayed even if the state_color is undefined.

I need to check that, but that sounds buggy to me.

**Comment by spacegaier** [comment]
Reg. the last point: Did you mean `entities` and not `entity` card? Because for the latter everything behaves correctly on my end.

**Comment by piitaya** [comment]
We have rules like this in state-badge (enttiies), entity and button :
```
state_color || (domain === "light" && state_color !== false)
 ```
So : 
- state_color = undefined => light colored, others domain not colored
- state_color = true => all domain colored
- state_color = false => all domain not colored

3 states boolean 😅

**Comment by piitaya** [comment]
We have also differences between tile card and entities card : the off state is grey on tile card and it use the default color for others card.

**Comment by spacegaier** [comment]
> We have rules like this in state-badge (used by entities), entity and button :
> 
> ```
> state_color || (domain === "light" && state_color !== false)
> ```
> 
> So :
> 
>     * state_color = undefined => light colored, others domains not colored
> 
>     * state_color = true => all domains colored
> 
>     * state_color = false => all domains not colored
> 
> 
> 3 states boolean 😅

Yes, I just saw that coding as well, but I think that has been that way for a long time, so no sense in changing that now. Probably not ideal, but the user can have full control here and with this PR we show the same consistent colors.

**Comment by spacegaier** [comment]
> We have also differences between tile card and entities card : the off state is grey on tile card and it use the default color for others card.

Yes I mentioned that in my PR description. I personally don't like that, since I strive for consistency to clearly communicate the state to the user. 

One could argue that gray is clearer here for lights, since the default blue could on first glance mean a blue LED. However, blue was always used as the default off state (and still is with the current theming changes). So the tile card goes against the status quo here (with the exclamation mark icon to differentiate for unknown/unavailable), so again inconsistent 😬 .

**Review by piitaya — DISMISSED** [review]

File: src/panels/lovelace/cards/hui-entity-card.ts (hunk 1 above)
Comment (piitaya) [reply src/panels/lovelace/cards/hui-entity-card.ts:217]: Yep we can change that in another PR 🙂

File: src/panels/lovelace/cards/hui-entity-card.ts [hunk 2]
```diff
@@ -196,13 +197,24 @@ export class HuiEntityCard extends LitElement implements LovelaceCard {
...
     }
     return undefined;
   }
 
+  private _computeBrightness(stateObj: HassEntity | LightEntity): string {
+    if (stateObj.attributes.brightness && stateActive(stateObj)) {
+      const brightness = stateObj.attributes.brightness;
+      return `brightness(${(brightness + 245) / 5}%)`;
+    }
+    return "";
```
Comment (piitaya) [inline src/panels/lovelace/cards/hui-entity-card.ts:215]:
```suggestion
    return undefined;
```

File: src/panels/lovelace/cards/hui-entity-card.ts [hunk 3]
```diff
@@ -196,13 +197,24 @@ export class HuiEntityCard extends LitElement implements LovelaceCard {
...
+      return `rgb(${stateObj.attributes.rgb_color.join(",")})`;
+    }
     const iconColor = stateColorCss(stateObj);
     if (iconColor) {
       return `rgb(${iconColor})`;
     }
     return undefined;
   }
 
+  private _computeBrightness(stateObj: HassEntity | LightEntity): string {
```
Comment (piitaya) [inline src/panels/lovelace/cards/hui-entity-card.ts:210]:
```suggestion
  private _computeBrightness(stateObj: HassEntity | LightEntity): string | undefined {
```

Comment (piitaya) [reply src/panels/lovelace/cards/hui-entity-card.ts:210]: We can also do the same in button card.

**Review by piitaya — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]

**Pull request merged**